setup.py

```diff
@@ -287,46 +286,75 @@ def write_version(filename: str = os.path.join(*["airflow", "git_version"])):
     'jira',
     'mongomock',
     'moto==1.3.5',
+    'mypy==0.720',
     'nose',
     'nose-ignore-docstring==0.2',
     'nose-timer',
     'parameterized',
-    'paramiko',
     'pre-commit',
     'pylint~=2.3.1',  # to be upgraded after fixing https://github.com/PyCQA/pylint/issues/3123
                       # We should also disable checking docstring at the module level
-    'pysftp',
-    'pywinrm',
-    'qds-sdk>=1.9.6',
     'rednose',
     'requests_mock',
     'yamllint'
 ]
+
+devel = sorted(devel + doc)
+
 ############################################################################################################
 # IMPORTANT NOTE!!!!!!!!!!!!!!!
 # IF you are removing dependencies from the above list, please make sure that you also increase
 # DEPENDENCIES_EPOCH_NUMBER in the Dockerfile
 ############################################################################################################
 
-if PY3:
-    devel += ['mypy==0.720']
-else:
-    devel += ['unittest2']
-
 devel_minreq = devel + kubernetes + mysql + doc + password + cgroups
 devel_hadoop = devel_minreq + hive + hdfs + webhdfs + kerberos
-devel_all = (sendgrid + devel + all_dbs + doc + samba + slack + oracle +
-             docker + ssh + kubernetes + celery + redis + gcp + grpc +
-             datadog + zendesk + jdbc + ldap + kerberos + password + webhdfs + jenkins +
-             druid + pinot + segment + snowflake + elasticsearch + sentry +
-             atlas + azure + aws + salesforce + cgroups + papermill + virtualenv)
 
-# Snakebite & Google Cloud Dataflow are not Python 3 compatible :'(
-if PY3:
-    devel_ci = [package for package in devel_all if package not in
-                ['snakebite>=2.7.8', 'snakebite[kerberos]>=2.7.8']]
-else:
-    devel_ci = devel_all
+all_packages = (
+    async_packages +
+    atlas +
+    all_dbs +
+    aws +
+    azure +
+    celery +
+    cgroups +
+    datadog +
+    dask +
+    databricks +
+    datadog +
+    docker +
+    druid +
+    elasticsearch +
+    gcp +
+    grpc +
+    flask_oauth +
+    jdbc +
+    jenkins +
+    kerberos +
+    kubernetes +
+    ldap +
+    oracle +
+    papermill +
+    password +
+    pinot +
+    redis +
+    salesforce +
+    samba +
+    sendgrid +
+    sentry +
+    segment +
+    slack +
+    snowflake +
+    ssh +
+    statsd +
+    virtualenv +
+    webhdfs +
+    winrm +
+    zendesk
+)
+
+# Snakebite is not Python 3 compatible :'(
+all_packages = [package for package in all_packages if not package.startswith('snakebite')]
```

Comment:
We should do this better and delcare those deps as `'snakebite;python_version<"3"'` etc.

Reply:
Yep. Will try to improve it.

Reply:
@ashb - That pretty much means that we should remove hdfs support in general because it needs snakebite (it's the only dependency it needs). Also kerberos has snakebite extra installed. 

Should we do it now? 

I can remove the snakebite dependencies and see if all pylint/flake etc complain (they will probably) or remove hdfs altogether ? (since it does not work maybe we should indeed remove it completely now?)

Reply:
There's an open issue for upgrading/replacing snakebite.

Removing Kerberos: eek no definetly not. Soem enterprises need kerberos support. If it's broken on Py3 then we need to fix that before 2.0.0 can be released.

Reply:
We are not removing kerberos as such but rather snakebite's kerberos extension in this change. I started parallel discussion in #5659 and we seem to have quite a taskforce to add better HDFS support. I'd rather leave it from dependencies now and let it be reimplemented. I think this might be cleaner solution than leaving "not-working" version in.